### ISSUE TITLE
update_page_form_field_values causes an "list assignment out of range" exception

### ISSUE THREAD

**Issue by MitchiLaser** [py-pdf/pypdf#1964]
I am using pypdf to fill in form values. I wrote a script which tracks all of my working time and fills out a form inside a PDF where I have to note all of them. Since pypdf version 3.10.0 the software crashes with the following error message:

```
Traceback (most recent call last):
  File "/ ... /test.py", line 70, in <module>
    pdf_writer.update_page_form_field_values(pdf_reader.pages[0], {field: form_data[field]})
  File "/ ... /venv/lib/python3.11/site-packages/pypdf/_writer.py", line 959, in update_page_form_field_values
    self._update_text_field(writer_annot)
  File "/ ... /venv/lib/python3.11/site-packages/pypdf/_writer.py", line 900, in _update_text_field
    self._objects[n - 1] = dct
    ~~~~~~~~~~~~~^^^^^^^
IndexError: list assignment index out of range

```

## Replicate the error

I wrote a script for you which is a cutout of my software (it was stored in a file named "test.py" as you can see in the stack trace above. It downloads the same pdf from the internet and tries to edit the form inside.

```python
a#!/usr/bin/env python3

MILOG_FORM_URL = r"https://www.pse.kit.edu/downloads/Formulare/KIT%20Arbeitszeitdokumentation%20MiLoG.pdf"

import sys
import os
from pypdf import PdfReader, PdfWriter
import tempfile
import requests
from pprint import pprint

#########################################

form_data = {
    'Std' : 20.0,
    'Summe' : 20.0,
    'monatliche SollArbeitszeit' : 20.0,
    'Urlaub anteilig' : 0,
    'Übertrag vom Vormonat' : 0, 
    'Übertrag in den Folgemonat' : 0, 
    'Stundensatz' : "12 €", 
    'Personalnummer' : "12345678", 
    'OE' : "Computer Science",
    'GF' : "Max Mustermann",
    'abc' : "04",
    'abdd' : "2023",
    'undefined' : '', 
    'Ich bestätige die Richtigkeit der Angaben' : "01.05.2023",
    'hhmmRow1' : '09:00',
    'hhmmRow1_2': '18:00',
    'hhmmRow1_3': '02:00',
    'hhmmRow1_4': '07:00',
    'hhmmRow2': '08:00',
    'hhmmRow2_2': '16:00',
    'hhmmRow2_3': '02:00',
    'hhmmRow2_4': '06:00',
    'hhmmRow3': '11:00',
    'hhmmRow3_2': '20:00',
    'hhmmRow3_3': '02:00',
    'hhmmRow3_4': '07:00',
    'ttmmjjRow1': '12.07.23',
    'ttmmjjRow2': '18.07.23',
    'ttmmjjRow3': '24.07.23'
}



print("\nForm Data:")
pprint(form_data)

with tempfile.TemporaryFile() as temp:

    # download online form and store it in a temp file
    try:
        r: requests.Response = requests.get(MILOG_FORM_URL, allow_redirects=True)
    except Exception as e:
        print(f"Exception when downloading PSE-Hiwi Formular -> {e}\n")
        sys.exit(os.EX_UNAVAILABLE)

    temp.write(r.content)
    temp.seek(0)    # move cursor back to the beginning of the file

    pdf_reader = PdfReader(temp)
    pdf_writer = PdfWriter()

    fields = pdf_reader.get_form_text_fields()  # get the field names from the form in the pdf
    for field in fields:                    # fill out all the fields in the form
        if field in form_data:
            pdf_writer.update_page_form_field_values(pdf_reader.pages[0], {field: form_data[field]})

    pdf_writer.add_page(pdf_reader.pages[0])    # put form content and page in a pdf-writer object

with open("output.pdf", 'wb') as output_file:    # write file
    pdf_writer.write(output_file)
```

With pypdf version 3.9.1 it still works but with pypdf version 3.10.0 and never the software does not work anymore. The script also needs to have the package `requests` installed in order to work.

From what I understand this has to do something with the fox of #1864, the software crashes at line 9000 in the file _writer.py which has been modified since the last working release as you can see here: [https://github.com/py-pdf/pypdf/compare/3.9.1...3.10.0#](https://github.com/py-pdf/pypdf/compare/3.9.1...3.10.0#)

**Comment by pubpub-zz** [comment]
as the PdfWriter works on fields, you have first to setup the pdf_writer with `pdf_writer.append()` and then update fields in the PdfWriter:
```
(...)
    pdf_reader = PdfReader(temp)
    pdf_writer = PdfWriter(clone_from=pdf_reader) # to copy everything else pdf_writer= PdfWriter();pdf_writer.append(pdf_reader)

    fields = pdf_reader.get_form_text_fields()  # get the field names from the form in the pdf
    for field in fields:                    # fill out all the fields in the form
        if field in form_data:
            pdf_writer.update_page_form_field_values(pdf_writer.pages[0], {field: form_data[field]})

with open("output.pdf", 'wb') as output_file:    # write file
    pdf_writer.write(output_file)
```

**Comment by MitchiLaser** [comment]
I tried your suggestion but the error stays the same.
Furthermore as I mentioned I don't think that I used the package wrongly because up to version 3.10.0 everything worked fine. Since version 3.10.0 the program crashes. Even the documentation tells that the first argument of `update_page_form_field_values` is in my script comming from `pdfReader` instead of `pdfWriter`, shouldn't this work the same way? (Documentation link: [https://pypdf.readthedocs.io/en/latest/user/forms.html](https://pypdf.readthedocs.io/en/latest/user/forms.html)
